pkg/provider/provider.go

```diff
@@ -222,3 +282,61 @@ func ParsePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
 	}
 	return rsaPrivateKey, nil
 }
+
+type Result struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
+}
+
+func GetOauthData(refreshToken, redirectUrl string) url.Values {
+	data := url.Values{}
+	data.Set("grant_type", "refresh_token")
+	data.Set("refresh_token", refreshToken)
+	data.Set("redirect_uri", redirectUrl)
+	return data
+}
+
+func GetOauthRequest(dataContent io.Reader, endPoint, clientId, clientSecret string) (*http.Request, error) {
+	request, err := http.NewRequest("POST", endPoint, dataContent)
+	if err != nil {
+		return nil, errors.Wrap(err, "Request to the endpoint could not be completed")
+	}
+	request.SetBasicAuth(clientId, clientSecret)
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+	return request, nil
+
+}
+
+func GetOauthAccessToken(
+	endPoint,
+	client_id,
+	client_secret string,
+	data url.Values) (string, error) {
+
+	client := &http.Client{}
+	request, err := GetOauthRequest(strings.NewReader(data.Encode()), endPoint, client_id, client_secret)
+	if err != nil {
+		return "", errors.Wrap(err, "Oauth request returned an error:")
+	}
+
+	var result Result
+
+	response, err := client.Do(request)
+	if err != nil {
+		return "", errors.Wrap(err, "Response status returned an error:")
+	}
+	if response.StatusCode != 200 {
+		errors.New(fmt.Sprintf("Response status code: %s: %s", strconv.Itoa(response.StatusCode), http.StatusText(response.StatusCode)))
```

Comment:
Missed this one. Make sure to return the error here like you have done in the other places

Reply:
🤦 just reviewed and addressed.

Reply:
the latest change is `	if response.StatusCode != 200 {
		return "", errors.New(fmt.Sprintf("Response status code: %s: %s", strconv.Itoa(response.StatusCode), http.StatusText(response.StatusCode)))
	}` https://github.com/gjv9491/terraform-provider-snowflake/blob/gjv9491/oauth_refresh_token/pkg/provider/provider.go